### PR TITLE
[INLONG-4251][GitHub] Improve GitHub configuration in `.asf.yaml`

### PR DESCRIPTION
**Pull request by shink**
fix: #4251

### Motivation

There are some improvable configuration in `.asf.yaml`, such as `Branch protection`, and so on.
Please check out https://cwiki.apache.org/confluence/exportword?pageId=127405038

### Modifications

- add `Branch protection` configuration in `.asf.yaml`

### CODE REVIEW

**Review by dockerzhang — APPROVED** [review]

File: .asf.yaml [hunk 1]
```diff
@@ -46,6 +46,43 @@ github:
...
+    release-0.5.0: {}
+    release-0.6.0: {}
+    release-0.7.0: {}
+    release-0.8.0: {}
+    release-0.9.0: {}
+    release-0.10.0: {}
+    release-0.11.0: {}
+    release-0.12.0: {}
+    release-1.0.0: {}
+    release-1.1.0: {}
```
Comment (dockerzhang) [inline .asf.yaml:85]: list all release branch is not convenient to maintain

Comment (dockerzhang) [reply .asf.yaml:85]: Can regular matching be used here?

Comment (healchow) [reply .asf.yaml:85]: @shink PTAL, tks.

**Review by healchow — CHANGES_REQUESTED** [review]

File: .asf.yaml (hunk 1 above)
Comment (shink) [reply .asf.yaml]: > list all release branch is not convenient to maintain

@dockerzhang Removed.

Comment (shink) [reply .asf.yaml]: > Can regular matching be used here?

There is more info on this on the [PR1678](https://github.com/apache/infrastructure-puppet/pull/1678) that implemented this cool feature. The regular matching is not supported now.

**Review by healchow — APPROVED** [review]

**Pull request merged**